src/abstract/Configurable.ts

```diff
@@ -0,0 +1,30 @@
+export abstract class Configurable<ConfigType extends object = {}> {
+    private readonly config: ConfigType;
+
+    protected constructor(config?: Partial<ConfigType>) {
+        this.config = {
+            ...(this.getDefaultConfig?.() ?? {}),
+            ...config
+        } as ConfigType;
+    }
+
+    public setConfigValue<K extends keyof ConfigType>(
+        key: K,
+        value: ConfigType[K]
+    ) {
+        this.config[key] = value;
+    }
+
+    public getConfigValue<K extends keyof ConfigType>(
```

Comment:
issue: In the interest of ease-of-typing, could we omit the getter and setter?

Reply:
Should have a getter and not have a setter, I only added the setter because we do some weird things in tests to change config values, which really should not happen in general as the config should be set once and read only and immutable